src/ProjectOrigin.PedersenCommitment/Ristretto/Oracle.cs

```diff
@@ -0,0 +1,48 @@
+namespace ProjectOrigin.PedersenCommitment.Ristretto;
+using System.Text;
+
+class Oracle
+{
+
+    private List<byte[]> messages = new List<byte[]>();
+
+    public Oracle(byte[] label)
+    {
+        messages.Add(label);
+    }
+
+    public void Domain(String domain)
+    {
+        var bytes = Encoding.UTF8.GetBytes("test");
+        messages.Add(bytes);
+    }
+
+
+    public void Add(params Point[] points)
+    {
+
+        foreach (Point p in points)
+        {
+            messages.Add(p.Compress()._bytes);
+        }
+    }
+
+    public Scalar Challenge()
```

Comment:
This needs to recompute the current state of the oracle every time we need a challenge (cf. top comment)

Reply:
I'm not sure I understand? The points are only compressed when adding them as messages, the challenge method only packs them into a single byte string and parses that along.

Reply:
Ah sure. But if it comes to that I would probably just use the merlin package then, since the complexity and performance becomes more of an issue.